packages/core/src/services/chains/ProviderProcessor/processStreamText.ts

```diff
@@ -0,0 +1,64 @@
+import { StreamCommonData } from '../../../events/handlers'
+import { AIReturn } from '../../ai'
+import { saveOrPublishProviderLogs } from './saveOrPublishProviderLogs'
+
+async function processResponse({
+  aiResult,
+  commonData,
+}: {
+  commonData: StreamCommonData
+  aiResult: Awaited<AIReturn<'text'>>
+}) {
+  return {
+    documentLogUuid: commonData.documentLogUuid,
+    text: await aiResult.data.text,
+    usage: await aiResult.data.usage,
+    toolCalls: (await aiResult.data.toolCalls).map((t) => ({
```

Comment:
all these promises can throw what happens if they do

Reply:
I'm not sure they fail 
![image](https://github.com/user-attachments/assets/2117f70e-f155-49c0-a940-1f7837247f24)

Reply:
The response finish always unless is a network error between our server and the AI provider

![image](https://github.com/user-attachments/assets/3a444f6c-a495-4230-8814-2de981ab6916)